addons/stock/models/product.py

```diff
@@ -120,7 +120,7 @@ def _compute_valid_ean(self):
     @api.depends('stock_move_ids.product_qty', 'stock_move_ids.state', 'stock_move_ids.quantity')
     @api.depends_context(
         'lot_id', 'owner_id', 'package_id', 'from_date', 'to_date',
-        'location', 'warehouse',
+        'location', 'warehouse', 'company'
```

Comment:
why did you change the `depends_context` ? You are not calling `companies.ids` from the context but from the env, so it works by itself. Imo, there's no need for that.
```suggestion
        'location', 'warehouse'
```

Reply:
Actually, there's a special treatment for the `company` key (see [1]). The cache will use the value from the `env` instead of the `context`.

However, after writing the test, I noticed that this doesn't work if multiple companies are selected (since the cache only uses the first one). As a result I used the `allowed_company_ids` key instead (which considers all selected companies).

[1] https://github.com/odoo/odoo/blob/823f30f4fd01667e37fc90fbbd795c3e5e857103/odoo/api.py#L819-L820